Assets/FeatureVoid/VoidCollider.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class VoidCollider : MonoBehaviour {
+
+    private int heightRespawn = 0;
+    // This script repositions the mecha whenever he falls into the void
+    // A PlatforMemorizer Script must be attached to the mecha beforehand
+
+    void OnCollisionEnter(Collision collision)
+    {
+        GameObject other = collision.collider.gameObject;
+        if (other.tag == "MechaBody")
+        {
+            // When the Mecha falls, reposition it over the last platform it landed on
+            GameObject lastPlatform = other.GetComponent<MechaController>().LastPlatform;
```

Comment:
You are here doing operations on the mecha from a script attached to the void colliders, there is clearly an issue. You'd better do all these operations directly in the mecha controller, if the mecha collides with a `Void` collider.

Reply:
I did fix this. Also, some code lines appeared to be duplicated because I forgot to delete some obsolete files before committing last time. My bad.